### PR TITLE
Add dcadec formula

### PR DESCRIPTION
**Pull request by rcombs**

### CODE REVIEW

File: Library/Formula/dcadec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Dcadec < Formula
+  desc "DTS Coherent Acoustics decoder with support for HD extensions"
+  homepage "https://github.com/foo86/dcadec"
+  url "https://github.com/foo86/dcadec/archive/v0.1.0.tar.gz"
+  sha256 "52a68a63f79ae50e5e5fd9c631a715e9feeed5d2b4a8442f0b9208692012a43f"
+  head "https://github.com/foo86/dcadec.git"
+
+  def install
+    system "make", "all"
+    unless File.exist?("test/samples/README")
```
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: When would this file exist/not exist?

File: Library/Formula/dcadec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      system "git", "clone", "https://github.com/foo86/dcadec-samples.git", "test/samples"
+    end
+    system "make", "check"
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "#{bin}/dcadec", "-h"
```
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: Would be good to actual encode a sample file.

File: Library/Formula/dcadec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  desc "DTS Coherent Acoustics decoder with support for HD extensions"
+  homepage "https://github.com/foo86/dcadec"
+  url "https://github.com/foo86/dcadec/archive/v0.1.0.tar.gz"
+  sha256 "52a68a63f79ae50e5e5fd9c631a715e9feeed5d2b4a8442f0b9208692012a43f"
+  head "https://github.com/foo86/dcadec.git"
+
+  def install
+    system "make", "all"
+    unless File.exist?("test/samples/README")
+      system "git", "clone", "https://github.com/foo86/dcadec-samples.git", "test/samples"
```
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: If you do this it should use a `resource` instead of `git clone`

File: Library/Formula/dcadec.rb (hunk 1 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: It's in the submodule, and doesn't exist when downloading the tarball.

File: Library/Formula/dcadec.rb (hunk 3 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: How is that done for a git repo?

File: Library/Formula/dcadec.rb (hunk 2 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: `make check` does that (relying on the sample files being present).

File: Library/Formula/dcadec.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: In that case it'll never exist. If you use `https://github.com/foo86/dcadec.git` as the URL and provide `:revision` and `:tag` it'll download the submodules for you.

File: Library/Formula/dcadec.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: This isn't a test suite but a verification that the formula was installed and compiled properly. Please install some sample files and run the command here too. Thanks!

File: Library/Formula/dcadec.rb (hunk 1 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: Does that lock `--HEAD` to always installing that revision, or just indicate which one the tar.gz comes from?

Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: Nope, `--HEAD` is still unlocked but will share the repository. Homebrew will just check it out to the right revision.

File: Library/Formula/dcadec.rb (hunk 2 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: I'm not sure how best to go about this. The package's build system has a built-in way to decode a number of provided samples (from that submodule URL) and check that the output was correct. I could duplicate that test logic in Ruby, but  it'd probably be easier to just checkout the repo and run `make check`. Would that be preferred over having `make check` run in the install step? (It'd probably be convenient to have some way of specifying that there's a test suite within the repo)

File: Library/Formula/dcadec.rb (hunk 1 above)
Comment (rcombs) [inline Library/Formula/dcadec.rb]: So does that replace the `.tar.gz` URL, then?

File: Library/Formula/dcadec.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: It's sufficient to just `.install` a sample file and run with `system` which will make sure it didn't e.g. crash.

Comment (rcombs) [inline Library/Formula/dcadec.rb]: Alright, should I remove the `make check` and submodule, then?

Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: It's fine to keep them too; think of it as unit vs integration tets.

File: Library/Formula/dcadec.rb (hunk 3 above)
Comment (bfontaine) [inline Library/Formula/dcadec.rb]: Use:

```rb
resource "some-name" do
  url "https://github.com/foo86/dcadec-samples.git"
end
```

Comment (mikemcquaid) [inline Library/Formula/dcadec.rb]: Except in this case it's a submodule so not needed.

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/f0c4fcf9216ee9fd8a64b0787078bb55703f40fb. Thank you for your contribution to Homebrew @rcombs; we appreciate it! :smiley_cat:

**Pull request closed without merging**